### PR TITLE
Account for presence of inapplicable extension methods of the same viability when binding arguments for error recovery in presence of single instance method candidate.

### LINKED ISSUE #5128: async keyword overly sensitive in VS2015
When I write a method that returns `Task` and apply thr `async` modifier, then the intellisense in VS2015 seems to have lots of problems. If anything in the method doesn't compile, then the intellisense is broken. This is often due to something later in the method with a syntax error – it causes intellisense for everything prior in the method to just now work (this used to be ok in VS2013).

### PR DESCRIPTION
**Pull request by AlekseyTs**
Fixes #5128.

@dotnet/roslyn-compiler Please review.

### CODE REVIEW

**Comment by AlekseyTs** [comment]
Test prtest/win/dbg/unit32 please

**Comment by gafter** [comment]
LGTM.

The added complexity to the internal APIs is unfortunate, but I don't see a better way.

File: src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs [hunk 1]
```diff
@@ -676,6 +678,7 @@ private static void CheckRestrictedTypeReceiver(BoundExpression expression, Comp
         /// <param name="methodGroup">Method group if the invocation represents a potentially overloaded member.</param>
         /// <param name="delegateTypeOpt">Delegate type if method group represents a delegate.</param>
         /// <param name="diagnostics">Diagnostics.</param>
+        /// <param name="extensionMethodsOfSameViabilityAreAvailable"/>
```
Comment (jaredpar) [inline src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs]: Description?

File: src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs [hunk 2]
```diff
@@ -965,7 +971,7 @@ private static NamedTypeSymbol GetDelegateType(BoundExpression expr)
                 methods = constructedMethods.ToImmutableAndFree();
             }
 
-            if (methods.Length == 1)
+            if (methods.Length == 1 && !extensionMethodsOfSameViabilityAreAvailable)
             {
                 // If there is only one method in the group, we should attempt to bind to it.  That includes
```
Comment (jaredpar) [inline src/Compilers/CSharp/Portable/Binder/Binder_Invocation.cs]: Should this comment be updated to reflect the change in the `if` condition?

**Comment by AlekseyTs** [comment]
@jaredpar Comments were added/adjusted per your feedback.

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler Please review, need a second sign off.

**Comment by VSadov** [comment]
LGTM

**Pull request merged**